inc/database/class-tmsc-media.php

```diff
@@ -0,0 +1,301 @@
+	 * @var string
+	 */
+	public $type = 'post';
+
+	/**
+	 * Constructor. Set this as a post migrateable.
+	 */
+	public function __construct() {
+		parent::__construct( $this->type );
+	}
+
+	/**
+	 * Get legacy ID
+	 * @return int
+	 *
+	 */
+	public function get_legacy_id() {
+		if ( ! empty( $this->raw->MediaMasterID ) ) {
+			return $this->raw->MediaMasterID;
+		}
+	}
+
+	/**
+	 * Get the last updated data hash.
+	 * @return mixed string|false
+	 */
+	public function get_last_updated_hash() {
+		if ( ! empty( $this->object ) ) {
+			return $this->get_meta( 'tmsc_last_updated', true );
+		}
+		return false;
+	}
+
+	/**
+	 * Set the last updated data hash.
+	 */
+	public function set_last_updated_hash() {
+		if ( ! empty( $this->raw ) ) {
+			$this->update_meta( 'tmsc_last_updated', tmsc_hash_data( $this->raw ) );
+		}
+	}
+
+	/**
+	 * Get title
+	 * @return string
+	 */
+	public function get_title(){
+		$title = ( empty( $this->raw->Title ) ) ? $this->raw->RawTitle : $this->raw->Title;
+		return apply_filters( 'tmsc_set_object_title', $title, $this->raw );
+	}
+
+	/**
+	 * Get date of publication
+	 * @return int unix timestamp
+	 */
+	public function get_pubdate(){
+		return apply_filters( 'tmsc_set_object_pubdate', time(), $this->raw );
+	}
+
+	/**
+	 * Get body
+	 * @return HTML
+	 */
+	public function get_body(){
+		return apply_filters( 'tmsc_set_object_body', '', $this->raw );
+	}
+
+	/**
+	 * Get post slug
+	 * @return string post slug
+	 */
+	public function get_post_name() {
+		return sanitize_title_with_dashes( $this->get_title() );
+	}
+
+	/**
+	 * Get post parent
+	 * @return integer parent post id
+	 */
+	public function get_post_parent() {
+		return ( empty( $this->post_parent ) ) ? 0 : $this->post_parent;
+	}
+
+	/**
+	 * Get post type
+	 * @return string
+	 */
+	public function get_post_type() {
+		return 'attachment';
+	}
+
+	/**
+	 * Get post status
+	 * @return string
+	 */
+	public function get_post_status() {
+		return 'inherit';
+	}
+
+	/**
+	 * Save the final post status
+	 * @return string
+	 */
+	public function save_final_object_status() {
+		$this->object->post_status = $this->get_post_status();
+		$this->update( $this->object );
+	}
+
+	/**
+	 * Get post object
+	 * @return WP_Post
+	 */
+	public function get_post() {
+		return $this->get_object();
+	}
+
+	/**
+	 * Update the post (used in after_save usually)
+	 */
+	public function update() {
+		wp_update_post( $this->object );
+	}
+
+	/**
+	 * Load an existing post if it exists.
+	 */
+	public function load_existing() {
+		$this->object = null;
+		// Check for existing post by legacy ID
+		$legacy_id = $this->get_legacy_id();
+		if ( ! empty( $legacy_id ) ) {
+			$existing_post = tmsc_get_object_by_legacy_id( $legacy_id );
+			if ( ! empty( $existing_post ) ) {
+				$this->object = $existing_post;
+			}
+		}
+	}
+
+	/**
+	 * Save this post
+	 * @return boolean true if successfully saved
+	 */
+	public function save() {
+			/*
```

Comment:
Is this function supposed to only return false, or is this commented out for debugging?